### PR TITLE
revised merge logic

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

**Comment by ymatsuda** [comment]
Please review. @raymondng @yingjieMiao

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala [hunk 1]
```diff
@@ -79,6 +80,9 @@ class KeepRepoImpl @Inject() (
 
 
   override def save(model: Keep)(implicit session: RWSession) = {
+    assert(model.isPrimary && model.state != KeepStates.DUPLICATE || !model.isPrimary && model.state != KeepStates.ACTIVE,
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:83]: Nice use of assert :+1:

**Comment by raymondng** [comment]
lgtm. thanks!

**Comment by yingjieMiao** [comment]
lgtm!

**Pull request merged**